db_stress_tool/db_stress_test_base.cc

```diff
@@ -693,9 +693,19 @@ void StressTest::OperateDb(ThreadState* thread) {
             snapshot, rand_column_family, column_family->GetName(),
             keystr,   status_at,          value_at,
             key_vec};
-        thread->snapshot_queue.emplace(
-            std::min(FLAGS_ops_per_thread - 1, i + FLAGS_snapshot_hold_ops),
-            snap_state);
+        auto hold_for = FLAGS_snapshot_hold_ops;
+        if (FLAGS_long_running_snapshots) {
+          // Hold 10% of snapshots for 10x more
+          if (thread->rand.OneIn(10)) {
+            hold_for *= 10;
+            // Hold 1% of snapshots for 100x more
+            if (thread->rand.OneIn(10)) {
+              hold_for *= 10;
+            }
+          }
+        }
+        auto release_at = std::min(FLAGS_ops_per_thread - 1, i + hold_for);
```

Comment:
Do not use `auto` when the type info cannot be deduced from the context easily. Here it is debatable since hold_for's type is known.